hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -39,6 +39,35 @@ define([
 
       _.extend($scope, _.cloneDeep($scope.model));
 
+      /**
+       * Maps Absence Types with the leave data from the Contract
+       * or sets default revision data if Contract leave data does not exists yet
+       *
+       * @param {Array} leaveData
+       * @param {Object} contractRevisionIdObj
+       */
+      function mapAbsenceTypesWithContractLeaveData (leaveData, contractRevisionIdObj) {
+        _.each($scope.leave, function (leaveType) {
+          _.extend(leaveType, leaveData
+            ? _.find(leaveData, { leave_type: leaveType.leave_type }) ||
+            contractRevisionIdObj : contractRevisionIdObj);
+        });
+      }
+
+      /**
+       * Maps Absence Types with the leave data from the Revision
+       *
+       * @param {Array} leaveEntity
+       * @param {Array} leaveData
+       */
+      function mapLeaveEntityWithRevisionLeaveData (leaveEntity, leaveData) {
```

Comment:
These two functions are basically the same (I'm really sorry for the placeholder names, but I'm not sure about their exact meaning):

```js
function foo (bar, baz, bax) {
  bax = bax || '';

  _.each(bar, function (leaveType) {
    _.extend(leaveType, baz
      ? _.find(baz, { leave_type: leaveType.leave_type }) || bax
      : bax);
  });
}
```

Reply:
Done.